fix: search page issue of cards showing in single column fixed

heymitali opened the pull request:
### What
My last [PR#347](https://github.com/openfoodfacts/openfoodfacts-explorer/pull/347) got merged which improved the product card design on homepage. While following the discussion on threads, I realised that it also impacted the layout of search page. Before my changes it was single column but product cards were stretched to full width, and in my PR I have modified the card's width. Because of which it is leaving empty space at the right side of the layout, which is not correct.

So in this PR, I have made this one line change to fix the search cards layout. Sorry for this mess up! And since this is my mistake, I would like to fix this as soon as possible, so that it won't affect other developments going on in this project.

In this PR, I am not doing changes in any other design, my intension is to just fix this bug as soon as possible.

And Thanks for raising this here: https://github.com/openfoodfacts/openfoodfacts-explorer/issues/399#issuecomment-2777361415

### Screenshot

This is show search results were looking before my last PR #347 got merged. Observe that it was single column design initially.
<img width="1344" alt="Screenshot 2025-04-04 at 7 34 46 PM" src="https://github.com/user-attachments/assets/3bbf318a-2e36-4fa1-8adc-4b72f7eea615" />

Search Results after my last PR merged. After modifying product card width, it is leaving empty space on right side.
<img width="1340" alt="Screenshot 2025-04-04 at 7 33 52 PM" src="https://github.com/user-attachments/assets/6548eed7-00d9-4bc5-8514-452cc9928175" />

Search Results after current PR's fix:
<img width="1460" alt="Screenshot 2025-04-13 at 10 29 26 AM" src="https://github.com/user-attachments/assets/c39c2b06-22a7-4dfc-b10e-60462005da6b" />

Skeleton Loading is also fixed in this PR:

https://github.com/user-attachments/assets/5a6ebe79-f0fd-41e6-b629-227ebad98558

Fixes bug(s)
#399

swastikCommits commented:
Hi @heymitali, I apologize for interrupting, but I believe it would be beneficial to address this later. Currently, we are focused on discussing on fixing the UI for the landing page in issue #426. 

If this PR is merged without aligning with the logic and design of the landing page, it could create unnecessary burden on @VaiTon. Later on, someone else will have to adjust the Search page to match the landing page's logic/design.

In my opinion, it would be best to keep this PR in draft status for now. If we find that we are unable to implement the logic to fix the landing page, then you would be free to proceed with this PR.

heymitali commented:
The fix that I did in this PR is independent of UI design of landing page. This is just related to broken search page product list alignment which occurred due to my last merged PR as mentioned in PR description. That's way, I am fixing it in this PR.

@VaiTon what do you think?

akashshetty1997 commented:
@heymitali  hey one tip in dark mode placeholder isnt visible clearly you can add this `opacity-70 dark:invert` this will make it more visible . 

Before
![image](https://github.com/user-attachments/assets/6008e23c-0eeb-4328-80c2-8832b5194ba1)

After 
![image](https://github.com/user-attachments/assets/378ec11d-34cb-4798-a102-6f1c7ef4a6b9)

heymitali commented:
Great catch @akashshetty1997 ! Thanks a lot for suggestion, I will push the changes.

heymitali commented:
I think the 'check for merge conflicts' step has a bug. The strange thing is that it’s showing a reference to another pull request and causing my pull request to fail. @teolemon can you please help in checking this. I am attaching screenshot for my PR's automation below.

Error message: "Issue #243 does not have label 'PR: ready to ship'. No need to remove."

![Screenshot 2025-04-05 at 12 59 30 PM](https://github.com/user-attachments/assets/16206e51-ef56-478e-93bf-d6813fbce0f6)

VaiTon commented:
@heymitali 

I'm open to merging this PR, at least until we decide for the better.

Please rebase on main.

VaiTon commented:
I see this right now, I guess the recent merge of #454 caused this

![immagine](https://github.com/user-attachments/assets/a9c30898-3986-4972-b787-b3444c94a09f)

swastikCommits commented:
Yeah exactly that is what I pinpointed, the changes due to #454 will create a inconsistency in the search page.
Good for us, we can fix it here.

heymitali commented:
I have took latest merge with main. Also fixed card layout and shimmer UI bug caused by #454 
@VaiTon please check now.

Shimmer UI Bug (Before this PR changes):
<img width="1350" alt="Screenshot 2025-04-10 at 12 59 53 PM" src="https://github.com/user-attachments/assets/0624f843-ee30-4031-a6fb-47f69b7ac26c" />

Fixed Shimmer UI Bug (Current PR):
<img width="1508" alt="Screenshot 2025-04-10 at 12 58 41 PM" src="https://github.com/user-attachments/assets/63e09f38-e4b2-4911-a7f0-a68dbc782812" />

Card Layout on Search page is also fixed.
<img width="1508" alt="Screenshot 2025-04-10 at 1 12 35 PM" src="https://github.com/user-attachments/assets/eea3de4d-acd3-4f51-bb98-bde32095dea7" />

heymitali commented:
@VaiTon can you please help in reviewing this PR.

VaiTon commented on src/routes/products/search/+page.svelte:
```diff
@@ -15,48 +15,54 @@
-			{#if result.page > 3}
-				<button class="btn btn-disabled join-item">...</button>
-			{/if}
+<div class="mt-8 flex w-full">
+	<div class="grid w-full grid-cols-1 gap-4 sm:grid-cols-2 lg:grid-cols-3 lg:gap-2 xl:grid-cols-3">
```
Why do we need two divs here? Can't we collapse?

VaiTon commented:
@heymitali the pagination should not be at the same height of the products

![immagine](https://github.com/user-attachments/assets/42c15b8f-0fa3-461e-bd54-a3280fd45096)

heymitali commented:
@VaiTon I have fixed it right away. This was because initially the pagination was part of parent div which contains all cards as well as pagination. And the pagination should be outside the card because of grid layout in parent div. I did the required changes in latest commit. Please check.

<img width="1501" alt="Screenshot 2025-04-13 at 11 11 54 PM" src="https://github.com/user-attachments/assets/d09c6d14-5a8e-44f6-92b0-99794b02ae3b" />
<img width="583" alt="Screenshot 2025-04-13 at 11 11 47 PM" src="https://github.com/user-attachments/assets/d42fbd34-0e74-4aa1-8634-185016ce4a9b" />
